### PR TITLE
Smbios cleanup

### PR DESCRIPTION
**Pull request by andreiw**
As discussed in the PRS call, here's my rework on the SMBIOS bits that were left orphaned over at riscv-smbios.

### CODE REVIEW

File: smbios.adoc [hunk 1]
```diff
@@ -46,50 +46,49 @@ The current version of the structure is 000Ah (v0.10).
 Bit 7:0 Minor revision +
 The newer revision of RISC-V Processor-specific Block
 Structure is backward compatible with older version of this structure.
-|02h|000Ah|Structure Length|Byte|110|Length of Processor-specific Data
-| 03h| 000Ah| Hart ID| DQWORD| Varies| The ID of this RISC-V Hart
-| 13h| 000Ah| Boot Hart| BYTE| Boolean| 1: This is boot hart to boot system +
+| 02h| 000Ah| Hart ID| DQWORD| Varies| The ID of this RISC-V Hart
```
Comment (adurbin-rivos) [inline smbios.adoc]: This comment is more about the version. Should we bump the version since we're changing fields?

File: smbios.adoc [hunk 2]
```diff
@@ -29,67 +29,79 @@ on top of cite:[SMBIOS], and is optional and recommended for BRS-B. Additional n
...
 The width (See below) of Machine Mode native base integer ISA supported by this RISC-V
 Hart
-| 3Ah| 000Ah| Reserved| BYTE| ENUM| Placeholder for Hypervisor Mode
-| 3Bh| 000Ah| Supervisor Mode native base integer ISA width (S-XLEN)| BYTE| ENUM|
+| 3Ah| 0100h| Reserved| BYTE| ENUM| Placeholder for Hypervisor Mode
+| 3Bh| 0100h| Supervisor Mode native base integer ISA width (S-XLEN)| BYTE| ENUM|
 The width (See below) of Supervisor Mode native base integer ISA supported by this RISC-V
 Hart
-| 3Ch| 0000Ah| User Mode native base integer ISA width (U-XLEN)| BYTE| ENUM| The
+| 3Ch| 00100h| User Mode native base integer ISA width (U-XLEN)| BYTE| ENUM| The
```
Comment (adurbin-rivos) [inline smbios.adoc]: an extra leading 0

File: smbios.adoc [hunk 3]
```diff
@@ -29,67 +29,79 @@ on top of cite:[SMBIOS], and is optional and recommended for BRS-B. Additional n
...
+The minor revision is to be incremented anytime new fields are added in a backwards-compatible manner. The major revision is to be incremented on backwards-incompatible changes.
+
+[cols="1,1,1,1,3", width=95%, align="center", options="header"]
+|===
+| Version | Bits 15:8+
+Major revision
+| Bits 7:0+
+Minor revision
+| Combined | Description
+| v1.0 | 01h | 00h | 0100h | First BRS-defined definition
```
Comment (adurbin-rivos) [inline smbios.adoc:71]: My other comment is now addressed.  Should we put this first at the beginning of the section like it was before?

File: smbios.adoc [hunk 4]
```diff
@@ -22,7 +22,51 @@ on top of cite:[SMBIOS], and is optional and recommended for BRS-B. Additional n
...
+| 02h| 0100h| Hart ID| QWORD| Varies| The ID of this RISC-V hart
+| 0Ah| 0100h| Boot hart| BYTE| Boolean| 1: This is the boot hart. +
+0: This is not the boot hart.
+| 0Bh| 0100h| Machine Vendor ID | QWORD| Varies| The vendor ID of this
+RISC-V hart
+| 13h| 0100h| Machine Architecture ID| QWORD| Varies| Base
+microarchitecture of the hart. Value of 0 is possible to indicate the field is
+not implemented. The combination of Machine Architecture ID and Machine Vendor
+ID should uniquely identify the type of hart microarchitecture that is implemented.
+| 1Bh| 0100h| Machine Implementation ID| QWORD| Varies| Unique encoding
```
Comment (vlsunil) [inline smbios.adoc:51]: While I understand this is a spec, I have an implementation question for these fields. These vendor ID/impl ID for each hart can be fetched using SBI call from S-mode FW only from the hart which executes it right? Typically, the SMBIOS creator FW like EDK2 runs with single CPU. How is it supposed to create these fields  each hart?

File: smbios.adoc [hunk 5]
```diff
@@ -22,7 +22,51 @@ on top of cite:[SMBIOS], and is optional and recommended for BRS-B. Additional n
...
+The structure is defined in a manner consistent with the DMTF specification
+language (cite:[SMBIOS]), and is valid for processors declared as
+architecture 07h (64-bit RISC-V) only.
+
+[cols="2,2,3,2,2,4", width=95%, align="center", options="header"]
+|===
+| Offset | Version | Name      | Length | Value   | Description
+| 00h| 0100h|Revision|WORD|Varies|See <<smbios-psd-ver>>.
+| 02h| 0100h| Hart ID| QWORD| Varies| The ID of this RISC-V hart
+| 0Ah| 0100h| Boot hart| BYTE| Boolean| 1: This is the boot hart. +
```
[the remark this answers is not in the document]
Comment (andreiw) [reply smbios.adoc:43]: I would agree with you that it isn't. I'll take it out.

File: smbios.adoc (hunk 3 above)
Comment (andreiw) [reply smbios.adoc:71]: Personally I think its fine where it is. There's a link to the section from the table so you can easily jump to it.

File: smbios.adoc (hunk 2 above)
Comment (andreiw) [reply smbios.adoc]: These fields are ultimately deleted in the patch set.

**Review by adurbin-rivos — APPROVED** [review]

**Review by vlsunil — APPROVED** [review]
Looks good to me. Thanks a lot for taking care of this.

Now, this spec being moved to BRS, an ECR to SMBIOS spec required to update the link it is pointing for table 44?

**Pull request merged**